### PR TITLE
THE HONKMOTHER COMES

### PR DESCRIPTION
**Pull request by ynot01**
## Intent of your Pull Request

# FINISHED, TESTED & FULLY FUNCTIONAL
but shitcode apparently

### Goals

- [x] Set up honk_act for objects to be converted

- [x] Give **Her** appearance

- [x] Remember to give credit to MerchantPixels

- [x] Create **Her**

- [x] Fix /objs not being converted on processing

- [x] Make only 20% of things in range convert per process

- [x] Fix objs already converted being converted again

- [x] Have **Her** spawn on clown ops victory at the location of the detonation

- [ ] fix shitcode (???)

- Make a nice looking effect for converting turfs (optional) (spriter needed)

#### The Honkmother
- Is the one true god
- Kind of smaller than the other gods
- Does not fight the other gods, only ~~sabotages~~ pranks them by turning their shit bananium
- Floats around like a one true god
- Spawns from the bananium fission explosive going off
- Turns all things near into their bananium counterparts
- Can summon clown mobs from ghosts
- Turns guns into Staff of the Honkmother

<details>
 <summary>Sprite courtesy of MerchantPixels</summary>

</details>

<details>
 <summary>Preview images</summary>

</details>

## Changelog

:cl:  ynot01, MerchantPixels
rscadd: Added The Honkmother, who spawns on Clown Ops victory
/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
@jcatjmeow theres no work in progress until im told why its unmergable and how to fix it

**Comment by jcatjmeow** [comment]
@ynot01 hows this

File: code/game/gamemodes/clown_ops/honkmother.dm [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		"clown/clownhulk",
+		"clown/clownhulk/chlown",
+		"clown/clownhulk/honcmunculus",
+		"clown/clownhulk/destroyer",
+		"clown/clownhulk/punisher",
+		"clown/mutant",
+		"clown/mutant/blob",
+		"clown/mutant/thicc")
+	var/MobType = text2path("/mob/living/simple_animal/hostile/retaliate/[pick(L)]")
+	var/mob/R = new MobType(get_turf(src))
```
Comment (alexkar598) [inline code/game/gamemodes/clown_ops/honkmother.dm]:
```suggestion
	var/mob/R = new pick(typesof(/mob/living/simple_animal/hostile/retaliate/clown))
```

File: code/game/gamemodes/clown_ops/honkmother.dm [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	R.visible_message("<span class='warning'>[R] awakens!</span>")
+	R.key = O.key
+
+/obj/structure/destructible/honkmother/Bump(atom/A)
+	var/turf/T = get_turf(A)
+	if(T == loc)
+		T = get_step(T, dir) // NOTHING WILL STAND IN THE WAY OF PRANKS
+	forceMove(T)
```
Comment (alexkar598) [inline code/game/gamemodes/clown_ops/honkmother.dm]: Set movement_type to UNSTOPPABLE instead and delete this proc

File: code/game/gamemodes/clown_ops/honkmother.dm [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	pixel_x = -128
+	pixel_y = -128
+	//She's immortal! Unkillable! Unmatched!
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	appearance_flags = 0
+	light_power = 0.7
+	light_range = 15
+	//Yellow, the color of bananas!
+	light_color = "#FFFF000"
+	//How far things are turned around Her are turned into bananium
```
Comment (alexkar598) [inline code/game/gamemodes/clown_ops/honkmother.dm]:
```suggestion
	///How far things around Her are turned into bananium
```

File: code/game/gamemodes/clown_ops/honkmother.dm [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		var/turf/T = I
+		if(prob(20))
+			T.honk_act()
+	for(var/I in circleviewturfs(src, round(convert_range * 0.5)))
+		var/turf/T = I
+		T.honk_act(TRUE)
+	for(var/O in range(round(convert_range / 1.2), src))
+		var/obj/T = O
+		if(prob(20))
+			T.honk_act()
```
Comment (alexkar598) [inline code/game/gamemodes/clown_ops/honkmother.dm]: Having two loops isnt ideal but its better than having 3 loops looping through everything in view

```suggestion
	for(var/I in circleviewturfs(src, round(convert_range * 0.5)))
		var/turf/T = I
		T.honk_act(TRUE)
		for(var/I in T)
			var/atom/A = I
			A.honk_act()
```

File: code/game/gamemodes/clown_ops/honkmother.dm [hunk 5]
```diff
@@ -0,0 +1,83 @@
+//HONK HONK HONK HONK HONK HONK HONK
```
Comment (alexkar598) [inline code/game/gamemodes/clown_ops/honkmother.dm]: Thats not a proper autodocs comment

**Review by alexkar598 — DISMISSED** [review]

File: code/game/gamemodes/clown_ops/honkmother.dm (hunk 5 above)
Comment (ynot01) [reply code/game/gamemodes/clown_ops/honkmother.dm]: doesnt autodocs require 3 slashes?

Comment (alexkar598) [reply code/game/gamemodes/clown_ops/honkmother.dm]: It does, but you also need to change the content to make sense...

**Comment by ynot01** [comment]
moving to another project, may reopen in the future

**Review by fluffe9911 — APPROVED** [review]
should be fine now

**Pull request merged**